### PR TITLE
Netherlands with school holidays

### PR DESCRIPTION
**Pull request by Flix6x**
Any interest in adding optional school holidays?

refs #

For information, read and make sure you're okay with the [Contributing guidelines](https://github.com/novafloss/workalendar/blob/master/contributing.md#adding-new-calendars).

- [x] Tests with a significant number of years to be tested for your calendar.
- [x] Docstrings for the Calendar class and specific methods.
- [ ] Use the ``workalendar.registry_tools.iso_register`` decorator to register your new calendar using ISO codes (optional).
- [x] Calendar country / label added to the README.rst file.
- [x] Changelog amended with a mention like: "Added ``<country>`` by ``@pseudo`` (#)". **Note** *Please do NOT change the version number here. It's the project maintainers' duty.*

- [x] Tests with a significant number of years to be tested for your calendar.
- [ ] Changelog amended with a mention describing your changes. **Note** *Please do NOT change the version number here. It's the project maintainers' duty.*

### CODE REVIEW

File: workalendar/tests/__init__.py [hunk 1]
```diff
@@ -13,11 +13,12 @@
 
 class CoreCalendarTest(TestCase):
     cal_class = Calendar
+    kwargs = {}
```
Comment (Flix6x) [inline workalendar/tests/__init__.py:16]: I had to find a way to pass the `region` to the test class. I'm not sure if there's a better way to do that. Here I made it generic to pass such variables, which might also be useful for others?

**Review by brunobord — CHANGES_REQUESTED** [review]
Mainly two concerns:

* you can remove the type annotations for more readability and being coherent with the rest of the codebase,
* you can improve test coverage by making sure that corner cases are tested + make sure that the all school holidays are covered.

Keep it up, it's an excellent job so far!

**Comment by Flix6x** [comment]
Thanks for the feedback. I made changes accordingly. I also refactored my tests a bit so that others should find it easier to set up tests for new school dates.

One more note of interest: for a calendar that includes school days, information is lost when transforming the list of holidays into a dictionary, using:
```
>>> dict(calendar.holidays(year))
```
This is because some days are both a school holiday and a national holiday (January 1st for example), whereas the resulting dict has unique date keys. To preserve all information when turning the list of date-name tuples into a dict, I used:
```
>>> holidays = {}
>>> for h in self.cal.holidays(year):
        holidays.setdefault(h[0], []).append(h[1])
```
Then holidays becomes a dictionary (still with unique date keys) in which the values are a list of names, like:
```
{datetime.date(2020, 1, 1): ['Christmas holiday', 'New year']}
```

**Comment by brunobord** [comment]
That looks fine fine fine, @Flix6x!
I was in the making of a new release and I've stopped to eventually integrate your latest branch. But even though there's no conflict here, I saw that the changelog entry was not positioned at the right place, because you've forked before the last release Spain...

Could you please rebase from master? if you can't or don't have time, I'd be glad to do it myself and add your great contribution to the next release!

**Comment by brunobord** [comment]
all right, hold your horses. Since your change introduces the `kwargs` arguments to the core class, I think I'll add it to the next major release...
I'll handle this if you don't mind and don't worry, your great contribution will be honored! ;o)

**Comment by Flix6x** [comment]
Okay, thanks for the heads-up.

FYI, I have another [feature branch](https://github.com/SeitaBV/workalendar/compare/netherlands-with-school-holidays...SeitaBV:netherlands-with-carnival) on hold to become a PR, which introduces the carnival holiday (and carnival school holidays), being relevant to the southern NL region. I am waiting for the present PR to be merged with master. Alternatively, I could merge that work into this PR. Let me know if you prefer the latter approach.

**Comment by brunobord** [comment]
@Flix6x I prefer to keep the scopes of the PRs as limited as possible, to make sure we don't have nasty side-effects.

The #569 is ready to be merged, and you may want to branch off master right after that :o)

**Comment by brunobord** [comment]
Netherlands extra calendars are now part of the newest (13.0.0) release:

* https://github.com/peopledoc/workalendar/releases/tag/13.0.0
* https://pypi.org/project/workalendar/13.0.0/

Happy upgrading!
